Patch taken from the Lua list. Original author appears to be Mike Pall.
http://lua-users.org/lists/lua-l/2008-02/msg00043.html

--- slnunico.c	2006-07-26 19:20:04.000000000 +0200
+++ slnunico.c-new	2008-02-01 17:53:30.000000000 +0100
@@ -1321,7 +1321,7 @@
 	/* register unicode itself so require("unicode") works */
 	luaL_register(L, SLN_UNICODENAME,
 		uniclib + (sizeof uniclib/sizeof uniclib[0] - 1)); /* empty func list */
-	lua_pop(L, 1);
+	//lua_pop(L, 1);
 	lua_pushinteger(L, MODE_ASCII);
 	luaI_openlib(L, SLN_UNICODENAME ".ascii", uniclib, 1);
 #ifdef SLNUNICODE_AS_STRING
@@ -1360,6 +1360,9 @@
 		}
 	}
 #endif
+
+        lua_settop(L, 2);
+        
 	return 1;
 }
 
